docs/features/source-generators.cookbook.md

```diff
@@ -284,6 +282,292 @@ public partial class UserClass : INotifyPropertyChanged
+    <IncludeBuildOutput>false</IncludeBuildOutput> <!-- Do not include the generator as a lib dependency -->
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Package the generator in the analyzer directory of the nuget package -->
+    <None Include="$(OutputPath)\$(AssemblyName).dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
+  </ItemGroup>
+```
+
+### Use functionality from NuGet packages
+
+**User Scenario:** As a generator author I want to rely on functionality provided in NuGet packages inside my generator.
+
+**Solution:** It is possible to depend on NuGet packages inside of a generator, but special consideration has to be taken for distribution.
+
+Any *runtime* dependencies, that is, code that the end users program will need to rely on, can simply be added as a dependency of the generator NuGet package via the usual referencing mechanism.
+
+For example, consider a generator that creates code that relies on `Newtonsoft.Json`. The generator does not directly use the dependency, it just emits code that relies on the library being referenced in the users compilation. The author would add a reference to `Newtonsoft.Json` as a public dependency, and when the user adds the generator package it will referenced automatically.
+
+The generator can check the compilation for the presence of the `Newtonsoft.Json` assembly and issue a warning or error if not present.
+
+```xml
+<Project>
+  <PropertyGroup>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild> <!-- Generates a package at build -->
+    <IncludeBuildOutput>false</IncludeBuildOutput> <!-- Do not include the generator as a lib dependency -->
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Take a public dependency on Json.Net. Consumers of this generator will get a reference to this package -->
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.1" />
+
+    <!-- Package the generator in the analyzer directory of the nuget package -->
+    <None Include="$(OutputPath)\$(AssemblyName).dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
+  </ItemGroup>
+</Project>
+```
+
+```C#
+using System.Linq;
+
+[Generator]
+public class SerializingGenerator : ISourceGenerator
+{
+    public void Execute(SourceGeneratorContext context)
+    {
+        // check that the users compilation references the expected library 
+        if(!context.Compilation.ReferencedAssemblyNames.Any(ai => ai.Name.StartsWith("Newtonsoft.Json")))
+        {
+            context.ReportDiagnostic(/*error or warning*/);
+        }
+    }
+
+    public void Initialize(InitializationContext context)
+    {
+    }
+}
+```
+
+However, any *compile-time* dependencies, that is, used by the generator while it is is running, must be packaged directly alongside the generator assembly inside the generator NuGet package. There are no automatic facilities for this, and you will need to manually specify the dependencies to include.
```

Comment:
Think this is a bit confusing because there are two interpretations of compile-time here: when the generator is compiled or when the code consuming the generator is compiled. Would be clearer to express this scenario as the generator execution.